test/integration/KNXClient.test.ts

```diff
@@ -0,0 +1,473 @@
+					}
+				})
+			})
+		}
+
+		try {
+			client = new KNXClient(
+				{
+					loglevel: 'trace',
+					hostProtocol: 'TunnelUDP',
+					sniffingMode: true,
+				},
+				(c: KNXClient) => {
+					const server = new MockKNXServer(
+						getMockToggleResponses(),
+						c,
+					)
+					server.createFakeSocket()
+				},
+			)
+
+			return await new Promise<void>((resolve, reject) => {
+				client.on(KNXClientEvents.error, (error) => {
+					reject(error)
+				})
+
+				client.on(KNXClientEvents.connected, async () => {
+					try {
+						// First toggle - ON
+						client.write(groupId, true, dpt)
+						// Advance by exact deltas from the mock
+						await clock.tickAsync(9) // First response delta
+						await clock.tickAsync(44) // Second response delta
+						await clock.tickAsync(670) // Third response delta
+						await waitForIndication(true)
+
+						// Second toggle - OFF
+						client.write(groupId, false, dpt)
+						await clock.tickAsync(7) // First response delta
+						await clock.tickAsync(43) // Second response delta
+						await clock.tickAsync(563) // Third response delta
+						await waitForIndication(false)
+
+						await client.Disconnect()
+						resolve()
+					} catch (error) {
+						reject(error)
+					}
+				})
+
+				client.Connect()
+				// Initial connection timing
+				clock.tickAsync(9) // Connect response
+				clock.tickAsync(20) // First heartbeat response
+			})
+		} finally {
+			clock.restore()
+		}
+	})
+
+	test('should handle long network disconnection leading to auto-disconnect', async () => {
+		const clock = sinon.useFakeTimers({
+			shouldAdvanceTime: true,
+			toFake: [
+				'setTimeout',
+				'clearTimeout',
+				'setInterval',
+				'clearInterval',
+				'Date',
+			],
+		})
+
+		try {
+			const events: string[] = []
+			let server: MockKNXServer
+			let disconnectReason = ''
+
+			// Store original constants
+			const originalStateTimeout =
+				KNX_CONSTANTS.CONNECTIONSTATE_REQUEST_TIMEOUT
+			const originalAliveTime = KNX_CONSTANTS.CONNECTION_ALIVE_TIME
+
+			// Override timeouts for test
+			KNX_CONSTANTS.CONNECTIONSTATE_REQUEST_TIMEOUT = 0.1 // 100ms instead of 10s
+			KNX_CONSTANTS.CONNECTION_ALIVE_TIME = 0.1 // 100ms instead of 60s
+
+			const client = new KNXClient(
+				{
+					loglevel: 'trace',
+					hostProtocol: 'TunnelUDP',
+					ipAddr: MockKNXServer.host,
+					ipPort: MockKNXServer.port,
+					connectionKeepAliveTimeout: 0.1, // This needs to match CONNECTION_ALIVE_TIME
+					localIPAddress: getDefaultIpLocal(),
+				},
+				(c: KNXClient) => {
+					server = new MockKNXServer(
+						getMockPacketsForDisconnectTest(),
+						c,
+					)
+
+					// Override error log to fail test when no matching response found
+					const originalError = server['error'].bind(server)
+					server['error'] = (message: string) => {
+						if (message.includes('No matching response found')) {
+							throw new Error(`MockKNXServer error: ${message}`)
+						}
+						originalError(message)
+					}
```

Comment:
change approach instead, emit an event when there is an error and intercept it here with server.on('error'). Remember that in order to do this mocksewrver should extend eventemitter and call super() on constructor. See how I did it on KnxClient